### PR TITLE
return 400 when encounter invalid encrypted key

### PR DESCRIPTION
**Pull request by watashiSHUN**
https://github.com/projectkudu/kudu/blob/6152ca9c6d5d40a0dca4b0f450b8659805457c75/Kudu.Services/Filters/FunctionExceptionFilterAttribute.cs#L29-L32
will do the conversion of 400 --> 500

this should address #2555

### CODE REVIEW

**Comment by watashiSHUN** [comment]
TODO need to verify that encryption key and decryption key used in test is consistent, ie: set environment variable `AzureWebEncryptionKey`

**Comment by davidebbo** [comment]
@watashiSHUN what's the status of this PR?

**Comment by watashiSHUN** [comment]
waiting for approval

File: Kudu.Core/Infrastructure/SecurityUtility.cs [hunk 1]
```diff
@@ -35,8 +35,15 @@ private static string GenerateSecretString()
...
-            var protector = DataProtectionProvider.CreateAzureDataProtector().CreateProtector(DefaultProtectorPurpose);
-            return protector.Unprotect(content);
+            try
+            {
+                var protector = DataProtectionProvider.CreateAzureDataProtector().CreateProtector(DefaultProtectorPurpose);
+                return protector.Unprotect(content);
+            }
+            catch (CryptographicException)
+            {
+                throw new FormatException($"unable to decrypt {content}, the key is either invalid or malformed");
```
Comment (davidebbo) [inline Kudu.Core/Infrastructure/SecurityUtility.cs]: Shouldn't the original CryptographicException be passed in as the inner exception?

            catch (CryptographicException e) {
                throw new FormatException($"unable to decrypt {content}, the key is either invalid or malformed", e);

**Comment by watashiSHUN** [comment]
rebased https://github.com/projectkudu/kudu/commit/94fccd0c4d4607fe5c1076b9aef0f1d588a27061

**Pull request closed without merging**